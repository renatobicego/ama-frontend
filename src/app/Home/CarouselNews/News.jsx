/* eslint-disable react/prop-types */
import Image from "next/image";
import Link from "next/link";

const News = ({news}) => {
    return(
        <Link href={news.href} role="link">
            <div className="relative h-full w-full">
                
                <Image
                    src={news.imgHref}
                    width={1500}
                    height={1500}
                    alt={`Imagen de noticia con tÃ­tulo ${news.title}`}
                    className="h-full w-full object-cover"
                    priority={true}
                />
                <div className="absolute inset-0 flex justify-center items-end h-full w-full bg-gradient-to-t from-title to-70%">
                    <div className="size-section text-left mb-[5%] lg:mb-[1%] ">
                        <h5 className="text-xs text-white md:text-base bg-tag-bg py-1 px-2 max-w-fit border-l-2 border-l-primary1 mb-2 font-title">
                            {news.category}
                        </h5>
                        <h2 className="mb-4 text-2xl md:text-3xl lg:text-4xl xl:text-5xl 2xl:text-6xl font-title font-semibold text-white">
                            {news.title}
                        </h2>
                        <h4 className="mb-12 text-sm md:text-lg lg:text-xl xl:text-2xl opacity-80 font-title text-white">
                            {news.subtitle}
                        </h4>
                    </div>
                </div>
            </div>
        </Link>
    )
}

export default News