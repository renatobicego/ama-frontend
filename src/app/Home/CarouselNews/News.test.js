import '@testing-library/jest-dom/extend-expect'
import News from './News'
import { cleanup, render, screen} from '@testing-library/react'

describe('<News />', () => {
    let news

    const newsMocked = {
        tag: 'Últimas Noticias',
        title: 'Borem ipsum dolor sit amet, consectetur adipiscing elit.',
        subtitle: 'Morem ipsum dolor sit amet, consectetur adipiscing elit. Nunc vulputate libero et velit interdum, ac aliquet odio mattis. Class',
        imgHref: '/test/test.jpg',
        href: '/noticias/titulo/test',
        category: 'Torneos'
    }

    beforeEach(() => {
        news = render(<News news={newsMocked}/>)
        
    })

    afterEach(() => {
        cleanup()
    })


    test('renders content', () => {
        expect(news.container).toMatchSnapshot()

        //Renders title and subtitle
        const title = screen.getByText(newsMocked.title)
        expect(title).toBeInTheDocument()
        const subtitle = screen.getByText(newsMocked.subtitle)
        expect(subtitle).toBeInTheDocument()

        // Renders img
        const img = screen.getByAltText(`Imagen de noticia con título ${newsMocked.title}`)
        expect(img).toBeInTheDocument()
    })

    test('redirection to news article', async() => {
        expect(news.container).toMatchSnapshot()

        //Get news href
        const linkElement = screen.getByRole('link')
        expect(linkElement.getAttribute('href')).toBe(newsMocked.href)
        
    })
})