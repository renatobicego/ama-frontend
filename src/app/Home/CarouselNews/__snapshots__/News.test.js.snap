// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<News /> redirection to news article 1`] = `
<div>
  <a
    href="/noticias/titulo/test"
    role="link"
  >
    <div
      class="relative h-full w-full"
    >
      <img
        alt="Imagen de noticia con título Borem ipsum dolor sit amet, consectetur adipiscing elit."
        class="h-full w-full object-cover"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="1500"
        src="/_next/image?url=%2Ftest%2Ftest.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Ftest%2Ftest.jpg&w=1920&q=75 1x, /_next/image?url=%2Ftest%2Ftest.jpg&w=3840&q=75 2x"
        style="color: transparent;"
        width="1500"
      />
      <div
        class="absolute inset-0 flex justify-center items-end h-full w-full bg-gradient-to-t from-title to-70%"
      >
        <div
          class="size-section text-left mb-[5%] lg:mb-[1%] "
        >
          <h5
            class="text-xs text-white md:text-base bg-tag-bg py-1 px-2 max-w-fit border-l-2 border-l-primary1 mb-2 font-title"
          >
            Torneos
          </h5>
          <h2
            class="mb-4 text-2xl md:text-3xl lg:text-4xl xl:text-5xl 2xl:text-6xl font-title font-semibold text-white"
          >
            Borem ipsum dolor sit amet, consectetur adipiscing elit.
          </h2>
          <h4
            class="mb-12 text-sm md:text-lg lg:text-xl xl:text-2xl opacity-80 font-title text-white"
          >
            Morem ipsum dolor sit amet, consectetur adipiscing elit. Nunc vulputate libero et velit interdum, ac aliquet odio mattis. Class
          </h4>
        </div>
      </div>
    </div>
  </a>
</div>
`;

exports[`<News /> renders content 1`] = `
<div>
  <a
    href="/noticias/titulo/test"
    role="link"
  >
    <div
      class="relative h-full w-full"
    >
      <img
        alt="Imagen de noticia con título Borem ipsum dolor sit amet, consectetur adipiscing elit."
        class="h-full w-full object-cover"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="1500"
        src="/_next/image?url=%2Ftest%2Ftest.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Ftest%2Ftest.jpg&w=1920&q=75 1x, /_next/image?url=%2Ftest%2Ftest.jpg&w=3840&q=75 2x"
        style="color: transparent;"
        width="1500"
      />
      <div
        class="absolute inset-0 flex justify-center items-end h-full w-full bg-gradient-to-t from-title to-70%"
      >
        <div
          class="size-section text-left mb-[5%] lg:mb-[1%] "
        >
          <h5
            class="text-xs text-white md:text-base bg-tag-bg py-1 px-2 max-w-fit border-l-2 border-l-primary1 mb-2 font-title"
          >
            Torneos
          </h5>
          <h2
            class="mb-4 text-2xl md:text-3xl lg:text-4xl xl:text-5xl 2xl:text-6xl font-title font-semibold text-white"
          >
            Borem ipsum dolor sit amet, consectetur adipiscing elit.
          </h2>
          <h4
            class="mb-12 text-sm md:text-lg lg:text-xl xl:text-2xl opacity-80 font-title text-white"
          >
            Morem ipsum dolor sit amet, consectetur adipiscing elit. Nunc vulputate libero et velit interdum, ac aliquet odio mattis. Class
          </h4>
        </div>
      </div>
    </div>
  </a>
</div>
`;
